src/dotnet-publish-iis/WebConfigTransform.cs

```diff
@@ -64,6 +64,25 @@ private static void TransformHttpPlatform(XElement httpPlatformElement, string a
             SetAttributeValueIfEmpty(httpPlatformElement, "stdoutLogEnabled", "false");
             SetAttributeValueIfEmpty(httpPlatformElement, "stdoutLogFile", logPath);
             SetAttributeValueIfEmpty(httpPlatformElement, "startupTimeLimit", "3600");
+
+            AddApplicationBase(httpPlatformElement);
+        }
+
+        private static void AddApplicationBase(XElement httpPlatformElement)
+        {
+            const string appBaseKeyName = "ASPNET_APPLICATIONBASE";
+
+            var envVariables = GetOrCreateChild(httpPlatformElement, "environmentVariables");
+            var appBaseElement = envVariables.Elements("environmentVariable").SingleOrDefault(e =>
+                string.Equals((string)e.Attribute("name"), appBaseKeyName, StringComparison.CurrentCultureIgnoreCase));
+
+            if (appBaseElement == null)
```

Comment:
This code is only modifying the web.config file - so there is a test verifying that the config is modified as [expected](https://github.com/aspnet/IISIntegration/pull/80/files#diff-2d39602621e11916e8439e3e56c8b466R17). ASPNET_APPLICATIONBASE is read somewhere in Hosting so I guess a testcase if any could be there (not sure if we really want to modify env variables during testing). Finally, the real E2E scenario for this involves dotnet publish, IIS, httpplatformhandler and hosting - I don't think we have any test like this but anytime you publish to iis you will invoke this code..